recipes/icu/all/conanfile.py

```diff
@@ -201,15 +208,15 @@ def build(self):
 
     @property
     def _data_filename(self):
-        vtag = self.version.split(".")[0]
+        vtag = Version(self.version).major
         return f"icudt{vtag}l.dat"
 
     @property
     def _data_path(self):
         data_dir_name = "icu"
         if self.settings.os == "Windows" and self.settings.build_type == "Debug":
             data_dir_name += "d"
-        data_dir = os.path.join(self.package_folder, "lib", data_dir_name, self.version)
+        data_dir = os.path.join(self.package_folder, "lib", data_dir_name, str(self.version))
```

Comment:
self.version is always a string AFAIK

Reply:
in 1.x is sometimes an object -- 2.0 should be a string